default.nix

```diff
@@ -1,5 +1,13 @@
 { pkgs ? import <nixpkgs> { } }:
 
 import ./lib {
-  inherit (pkgs) lib newScope;
+  inherit (pkgs) lib makeScopeWithSplicing';
+  otherSplices = {
+    selfBuildBuild = pkgs.pkgsBuildBuild;
```

Comment:
I don't really understand how this solves the issue. `selfBuildBuild` should be `buildBuild` variants of crane's packages, not the ones from nixpkgs. Splicing `cargo` will just pick `cargo` from `pkgsBuildBuild`, same as before.

Reply:
Oh, actually I guess it does work if you pass a toolchain overlayed over nixpkgs.

Reply:
yes this was exactly the problem, using `craneLib.overrideToolchain` would use `.overrideScope` to define a new `cargo` definition, except that wouldn't make it into the spliced packages themselves, so the version from `nixpkgs` was being selected (which was still "wrong" even if spliced "correctly")

Reply:
Nope, this doesn't work. `callPackage` will splice `cargo` here as `pkgsBuildBuild.cargo`.

I got a few idea on how to fix this in #648.